packages/nc-gui/components/dashboard/TreeView.vue

```diff
@@ -550,18 +565,40 @@ watch(
                       <template #title>{{ table.table_name }}</template>
                       <div class="flex items-center gap-2 h-full" @contextmenu="setMenuContext('table', table)">
                         <div class="flex w-auto" :data-testid="`tree-view-table-draggable-handle-${table.title}`">
-                          <MdiDragVertical
-                            v-if="isUIAllowed('treeview-drag-n-drop')"
-                            :class="`nc-child-draggable-icon-${table.title}`"
-                            class="nc-drag-icon text-xs hidden group-hover:block transition-opacity opacity-0 group-hover:opacity-100 text-gray-500 cursor-move"
-                            @click.stop.prevent
-                          />
-
-                          <component
-                            :is="icon(table)"
-                            class="nc-view-icon text-xs"
-                            :class="{ 'group-hover:hidden group-hover:text-gray-500': isUIAllowed('treeview-drag-n-drop') }"
-                          />
+                          <a-dropdown trigger="click" @click.stop destroy-popup-on-hide>
+                            <div @click.stop>
+                              <a-tooltip>
+                                <span v-if="table.meta?.icon" :key="table.meta?.icon">
+                                  <Icon :key="table.meta?.icon" class="text-xl" :icon="table.meta?.icon"></Icon>
+                                </span>
+                                <component
+                                  :is="icon(table)"
+                                  v-else
+                                  class="nc-view-icon w-5"
+                                  :class="{ 'group-hover:text-gray-500': isUIAllowed('treeview-drag-n-drop') }"
+                                />
+
+                                <template #title>Change icon</template>
+                              </a-tooltip>
+                            </div>
+                            <template #overlay>
+                              <GeneralEmojiIcons class="shadow bg-white p-2" @select-icon="setIcon($event, table)" />
+                            </template>
+                          </a-dropdown>
+
+                          <!--
```

Comment:
is it by design that the original drag-n-drop icon will be gone?

Reply:
Yes, earlier we used to toggle the table icon and now we can't toggle since the icon change menu triggers when clicking the table icon. So finally we made it draggable by holding it anywhere in the item.